codice/ddf
# DDF-4459 Export user selected results from a query

#### What does this PR do?
- Adds the ability to export a set of selected results from a query.
- Can export a single result or multiple results. With either type of selection a single file is created. 
- Updates query selection interface to be more user friendly. 

#### Who is reviewing it? 

@samuelechu 
@cantstoptheunk 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@djblue
@tbatie

#### How should this be tested?

- Ingest some metacards into the system. 
- Export a single metacard.
- Verify that metacard transformers appear in dropdown. 
- Verify that resulting document is correct. 
- Export a set of metacards. 
- Verify that the query response transformers appear in dropdown. 
- Verify that resulting document is correct. 

#### Any background context you want to provide?
- Build on top of the existing export functionality. 

#### What are the relevant tickets?
For Jira:
[DDF-4459](https://codice.atlassian.net/browse/DDF-4459)

#### Screenshots

<img width="1680" alt="screen shot 2019-02-15 at 8 45 02 am" src="https://user-images.githubusercontent.com/7646577/52871102-899f6400-3106-11e9-8077-1a05193020dd.png">
<img width="1680" alt="screen shot 2019-02-15 at 9 42 46 am" src="https://user-images.githubusercontent.com/7646577/52871103-899f6400-3106-11e9-9176-ccdd70514b1c.png">
<img width="1680" alt="screen shot 2019-02-15 at 9 42 55 am" src="https://user-images.githubusercontent.com/7646577/52871104-899f6400-3106-11e9-876b-0e8c89671547.png">
<img width="289" alt="screen shot 2019-02-13 at 9 17 03 am" src="https://user-images.githubusercontent.com/7646577/52871413-3e398580-3107-11e9-88f1-4340212122b6.png">

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
